### PR TITLE
TOURCMS-10026 add feature login with api

### PR DESCRIPTION
**Pull request by palisis-juancarlos**
and make minor changes about code format

### CODE REVIEW

File: controllers/Templates.php [hunk 1]
```diff
@@ -52,6 +53,39 @@ private function getIndex()
         return ($final);
     }
 
+    function getXMLFromValidation($url, $paramsFromGet)
```
Comment (palisis-juanra) [inline controllers/Templates.php]: This would be good to be in a service / library

File: controllers/Templates.php [hunk 2]
```diff
@@ -52,6 +53,39 @@ private function getIndex()
...
+            curl_close($ch);
+            return null;
+        }
+
+        curl_close($ch);
+
+        //Loads the XML response into a SimpleXMLElement object
+        $xml = simplexml_load_string($response);
+        if ($xml === false) {
+            echo "Failed loading XML\n";
```
Comment (palisis-juanra) [inline controllers/Templates.php]: This should not output errors

File: controllers/Templates.php [hunk 3]
```diff
@@ -52,6 +53,39 @@ private function getIndex()
...
+
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        $response = curl_exec($ch);
+
+        // Handle any errors
+        if (curl_errno($ch)) {
+            echo 'Curl error: ' . curl_error($ch);
```
Comment (palisis-juanra) [inline controllers/Templates.php]: This should not output errors

File: index.php [hunk 4]
```diff
@@ -18,13 +18,35 @@
...
+}
+
+if (isset($_POST['logout'])) {
+    session_start();
+    session_destroy();
+    header('Location: ' . $templates->getIndex() . '/login');
+}
+
+if ($page != 'login' && (!isset($_SESSION['logged']) || $_SESSION['logged'] != true)) {
+    error_log(print_r($_SESSION, true));
```
Comment (palisis-juanra) [inline index.php]: This error log is not useful, please remove it

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

File: index.php (hunk 4 above)
Comment (palisis-juancarlos) [reply index.php]: fixed

File: controllers/Templates.php (hunk 3 above)
Comment (palisis-juancarlos) [reply controllers/Templates.php]: fixed

File: controllers/Templates.php (hunk 2 above)
Comment (palisis-juancarlos) [reply controllers/Templates.php]: fixed, handled outisde of the method, in index file

File: controllers/Templates.php (hunk 1 above)
Comment (palisis-juancarlos) [reply controllers/Templates.php]: fixed, added to another individual class

**Review by palisis-juanra — APPROVED** [review]

**Pull request merged**